src/ui/compilationStatusView.ts

```diff
@@ -52,24 +66,52 @@ export default class CompilationStatusView {
     return view;
   }
 
+  public registerBefore38Messages(): void {
+    this.context.subscriptions.push(
+      this.languageClient.onCompilationStatus(params => this.compilationStatusChanged(params)),
+      this.languageClient.onVerificationStarted(params => this.verificationStarted(params)),
+      this.languageClient.onVerificationCompleted(params => this.verificationCompleted(params))
+    );
+  }
+
+  // Not used from 3.8
+  private verificationStarted(params: IVerificationStartedParams): void {
+    this.documentStatusMessages.set(
+      getVsDocumentPath(params),
+      { message: Messages.CompilationStatus.Verifying }
+    );
+    this.updateActiveDocumentStatus();
+  }
+
+  // Not used from 3.8
+  private verificationCompleted(params: IVerificationCompletedParams): void {
+    this.documentStatusMessages.set(
+      getVsDocumentPath(params),
+      { message: params.verified ? Messages.CompilationStatus.Verified : Messages.CompilationStatus.NotVerified }
+    );
+    this.updateActiveDocumentStatus();
+  }
+
   private documentClosed(document: TextDocument): void {
     this.documentStatusMessages.delete(document.uri.toString());
     this.updateActiveDocumentStatus();
   }
 
-  private areParamsOutdated(params: ICompilationStatusParams): boolean {
+  public areParamsOutdated(params: ICompilationStatusParams): boolean {
     const previous = this.documentStatusMessages.get(getVsDocumentPath(params));
     return previous?.version != null && params.version != null
        && previous.version > params.version;
   }
 
+
+
   private static readonly handledMessages = new Set([
     CompilationStatus.ResolutionStarted,
     CompilationStatus.ParsingFailed,
     CompilationStatus.ResolutionFailed,
     CompilationStatus.CompilationSucceeded ]);
 
-  private compilationStatusChanged(params: ICompilationStatusParams): void {
+  public compilationStatusChanged38(params: ICompilationStatusParams): void {
```

Comment:
Did you mean?
```suggestion
  public compilationStatusChanged37(params: ICompilationStatusParams): void {
```
I understand this is the old method that is going to be deprecated, so it should contain the version number that it is useful for.

Reply:
To be consistent with the comments "// Backwards compatibility for versions of Dafny <= 3.8.0" I think it's better to mention 3.8, but I've made the function name more explicit "compilationStatusChangedForBefore38"